includes/emails/class-wc-email.php

```diff
@@ -250,14 +250,23 @@ public function handle_multipart( $mailer ) {
 	/**
 	 * Format email string.
 	 *
-	 * @param mixed $string
+	 * @param mixed $string Text to replace placeholders in.
 	 * @return string
 	 */
 	public function format_string( $string ) {
-		// Legacy placeholders. @todo deprecate in 4.0.0.
+		$find    = array_keys( $this->placeholders );
+		$replace = array_values( $this->placeholders );
+
+		// If using legacy find replace, add those to our find/replace arrays first. @todo deprecate in 4.0.0.
+		if ( isset( $this->find, $this->replace ) ) {
+			$find    = array_merge( $this->find, $find );
+			$replace = array_merge( $this->replace, $replace );
+		}
+
+		// If using the older style filters for find and replace, ensure the array is associative and then pass through filters. @todo deprecate in 4.0.0.
 		if ( has_filter( 'woocommerce_email_format_string_replace' ) || has_filter( 'woocommerce_email_format_string_find' ) ) {
-			$legacy_find    = array();
-			$legacy_replace = array();
+			$legacy_find    = isset( $this->find ) ? $this->find : array();
```

Comment:
Can drop the `isset` here, as `find`/`replace` are initialied to empty array in this class.